samcli/lib/deploy/deployer.py

```diff
@@ -511,7 +515,7 @@ def wait_for_execute(
                 self._display_stack_outputs(outputs)
         except DeployStackOutPutFailedError as ex:
             # Show exception if we aren't deleting stacks
-            if not stack_operation == "DELETE":
+            if on_failure != FailureMode.DELETE:
```

Comment:
Can you briefly explain why we don't need to "or" this condition in for stack_operation == "DELETE" anymore?

Reply:
Yup, so previously I had added a waiter for stack_delete_complete, this meant that I had to add DELETE as a possible stack_operation. Now its been removed and the waiter is now located in the rollback_delete_stack() method, so we check the on_failure parameter instead.